docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+What should be a basic abstraction for representing a path of communication between the two?
+We use the term _connection_ to represent this path of communication.
+The purpose of the IBC connection handshake protocol, at the highest level of abstraction, is to enable two chains to establish a connection with each other [[ICS 003](#references)].
+Throughout the rest of this document, we will often use the abbreviation "CH" to stand for "connection handshake" problem (or protocol, depending on the context).
+
+#### Sequential problem statement
+
+Instead of _chains_, at this abstract level of specification we can reason in terms of _processes_.
+There are two processes in our problem statement, which we can label explicitly as an _initiator_ (or source) process and a _responder_ (or destination) process.
+
+
+###### Problem statement and guarantees
+A connection handshake allows, briefly, that two processes agree on a consistent view of their state, and each process allocates a new connection associated with this state.
+We identify the following requirements that the connection handshake should guarantee:
+
+- **[CH/Seq/1/Agreement]** If the initiator finishes the handshake, establishing a new connection `ci` with the responder, and the responder finishes the handshake obtaining the connection `cr`, then `ci == cr`.
+In other words, the two ends of the connection (one at the initiator and the other at the responder) are consistent.
+
+
+- **[CH/Seq/2/Termination]** Eventually, the two processes should both finish the handshake, establishing a connection.
+
+Requirement **[CH/Seq/1/Agreement]** represents the basic safety guarantee, while **[CH/Seq/2/Termination]** is the basic liveness guarantee we would like to have.
+
+The diagram below sketches the high-level problem statement.
+
+```
+                        IBC Connection
+     +-----------+    Handshake Protocol    +-----------+
+     |           |<-------- step 1 -------->|           |
+     | Initiator |    (execute handshake)   | Responder |
+     |           |                          |           |
+     |          _| (connection established) |_          |
+     |         |c|<------------------------>|c|         |
+     |          ▔|                          |▔          |
+     +-----------+                          +-----------+
+```
+
+
+## L2. Protocol specification / protocol view
+
+In this section we attempt to make the problem specification more concrete.
+We define what the processes are, their fault-model, as well as communication and synchrony assumptions.
+
+### 2.1 System model specification
+
+#### System model
+
+A **chain** is a state machine.
+Each chain may be replicated, i.e., consisting of a replicated state machine (RSM), or it may be a standalone process.
+In practice, there are certain requirements which each chain must satisfy (e.g., see [[ICS 024](#references)]).
+At this level of specification, however, we can assume each chain is a single **process**.
```

Comment:
It is maybe better talking about module instead of chain as we can have multiple modules from some chain establishing connections with modules from remote chain.

Reply:
I'm not sure we need to introduce the term "module" to speak about connection handshake at this level of abstraction. I think we need two types of entities, not more: we can think of them as **client** (this is the user of the connection handshake, who initiates the protocol; the client is the external observer) and **server** (whoever it is that exposes the abstraction we're specifying here and implements the CH protocol).

In the current version of specification:
- client = application, and 
- server = chain = essentially an arbitrary process.

Should I modify this (also considering Anca's comment below) to be the following?

- client = actor, and
- server = module.

This would be more aligned with IBC specs terminology so would be useful! But in my view introduces unnecessary details that belong to implementation.

Reply:
After thinking more about this, another option would be to have the following scheme:

- client = actor
- server = module
- and a chain is an external process, communicating with a module.

Reply:
After more thinking, I will incorporate the notion of "module" as a part of a chain. Hope this will resolve inconsistencies.